src/express/routes/articles-routes.js

```diff
@@ -56,23 +83,56 @@ articlesRouter.get(`/:id`, async (req, res) => {
 });
 
 articlesRouter.post(`/add`, upload.single(`photo`), async (req, res) => {
-  const {body} = req;
-
   const articleData = {
     // TODO тут будет поле photo, когда я пойму почему multer падает
     // TODO непонятно каким образом должны выбираться категории, кнопка в форме не работает, временно захардкодил котиков
+    categories: [1],
+    title: req.body.title,
+    announce: req.body.announcement,
+    fullText: req.body[`full-text`],
+  };
+
+  try {
+    await api.createArticle(articleData);
+    res.redirect(`/my`);
+  } catch (errors) {
+    const validationMessages = prepareErrors(errors);
+    const categories = await getAddArticleData();
+    res.render(`articles/post-add`, {categories, validationMessages});
+  }
+});
+
+articlesRouter.post(`/edit/:id`, upload.single(`avatar`), async (req, res) => {
+  const {body, file} = req;
+  const {id} = req.params;
+  const articleData = {
+    picture: file ? file.filename : body[`old-image`],
     category: [`Котики`],
     title: body.title,
     announce: body.announcement,
     fullText: body[`full-text`],
     createdDate: body.date,
   };
-
   try {
-    await api.createArticle(articleData);
+    await api.editOffer(id, articleData);
     res.redirect(`/my`);
-  } catch (error) {
-    res.redirect(`back`);
+  } catch (errors) {
+    const validationMessages = prepareErrors(errors);
+    const [article, categories] = await getEditArticleData(id);
+    res.render(`offers/ticket-edit`, {id, article, validationMessages, categories});
```

Comment:
```suggestion
    res.render(`articles/ticket-edit`, {id, article, validationMessages, categories});
```
так, тут что-то не то)
и так много где по проекту. подправь везде слово offer)